repos/system_upgrade/common/libraries/rpms.py

```diff
@@ -114,3 +115,36 @@ def check_file_modification(config):
     """
     output = _read_rpm_modifications(config)
     return _parse_config_modification(output, config)
+
+
+def get_leapp_packages(source_version=None, component=None, include_deps=False):
+    """
+    Get list of leapp packages.
+
+    :param source_version: a list or string specifying source_versions. If not defined then all packages
+        for current source_target_version will be returned.
+    :param component: a list or string specifying leapp components. If defined then only packages related
+        to the specific component will be returned.
+    :param include_deps: a flag to control whether deps meta packages are to be filtered out.
```

Comment:
First pass review: An api comment:  I think it would be better to add deps as either separate components or via a separate function.  Using `include_deps` here adds a new concept to understand and a matrix of  combinations to think about. If someone doesn't know what it does under the hood, just where it's used in the code, they might have these questions:

- Why does `get_leapp_packages(component=snactor, include_deps=True)` just return `snactor`?
- All the other parameters are to add additional information to the return, why does this one take things away? (Filter in documentation, not giving additional packages in the snactor case above)

Accessing them as components means you have two more component names (`framework-deps` and `repository-deps`) but you don't have to learn a new concept.

Writing a separate function lets you treat deps as wholly different concepts so you only encounter them in the case where you need to care about the deps packages too.  My impression is only one use case needs deps packages so this might make sense. However, for that use case, you then have to realize you need two use both functions to get what you want so it has a drawback too.  (You could also have a separate function to filter the deps but I think that's worse if the common case is to want results without deps).

Using separate components is what I personally would choose.

Reply:
We've moved include_deps to a separate function so this should now be resolved.